controllers/articlesController.js

```diff
@@ -37,60 +101,171 @@ export default class PostController {
+        const articles = all.map(each => orderArticle(each, id));
+        const articlesCount = articles.length;
+        return author || favorited || tag
+          ? next(articles)
+          : res.status(OK).json({
+              message: "Articles retrieved successfully",
+              articles,
+              articlesCount
+            });
+      }
+      return res.status(NOT_FOUND).json({
         message:
-          "Sorry, this is not working properly. We now know about this mistake and are working to fix it"
+          "No more articles found, you can also share your thoughts by creating an article"
       });
+    } catch (error) {
+      return res.status(INTERNAL_SERVER_ERROR).json({ message: SERVER_ERROR });
+    }
+  }
+
+  /**
+   * @description It helps the user to update any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async update(req, res) {
+    try {
+      const { slug } = req.params;
+      const { user: { id: userId } = {}, body } = req;
+      const article = await Article.findOne({ where: { slug, userId } });
+      if (article) {
+        const allowed = _.pick(body, Object.keys(articleSchema));
+        const valid = await articleValidator(
+          allowed,
+          _.pick(articleSchema, Object.keys(allowed))
+        );
+        const { tagsList = [], ...rest } = valid;
+        if (tagsList.length) {
+          tagsList.map(async t => {
+            const tag = await Tag.findOrCreate({ where: { name: t } }).spread(
+              tg => tg
+            );
+            await article.addTagsList(tag);
+          });
+        }
+        const updated = await article.update(rest, {
+          where: { id: article.id, userId },
+          returning: true,
+          limit: 1
+        });
+        const tags = await updated
+          .getTagsList({ attributes: ["name"] })
+          .map(t => t.name || null);
+        return res.status(CREATED).json({
+          article: {
+            ...updated.get(),
+            author: _.pick(await updated.getAuthor(), ["username", "bio", "image"]),
+            tagsList: tags
+          },
+          message: `Article updated successfully`
+        });
+      }
+      return res.status(UNAUTHORIZED).json({
+        message: `You can only update the article you authored`
+      });
+    } catch (error) {
+      return error.details
+        ? res.status(BAD_REQUEST).json({ message: error.details[0].message })
+        : res.status(INTERNAL_SERVER_ERROR).json({ message: SERVER_ERROR });
+    }
+  }
+
+  /**
+   * @description It helps the user to delete any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async deleteOne(req, res) {
+    try {
+      const { slug } = req.params;
```

Comment:
@Musigwa Are you validating the parameter's slug in order to avoid SQL injection?

Reply:
I think I've added the validation right after this message. Thanks for bringing that in.